Sources/SwiftCliCore/MetaDataBuilder .swift

```diff
@@ -0,0 +1,86 @@
+    lazy var offsetBottomLine: String = leftOffsetString + bottomLine
+    lazy var offsetStitchSymbols: String = transRowLeftOffsetString + leftOffsetString + stitchSymbols
+    lazy var totalRow: String = offsetStitchSymbols + offsetBottomLine
+
+}
+public class MetaDataBuilder {
+    public init() {}
+
+    public func gatherAllMetaData(stitchArray: [[String]]) -> [RowInfo] {
+        var allRowsMetaData = [] as [RowInfo]
+        let rowsNum = stitchArray.count
+        var upcomingleftOffset = 0
+        for row in 0..<(rowsNum) {
+            var newRowMetadata = makeRowMetadata(stitchRow: stitchArray[row], rowNumber: row)
+            newRowMetadata.leftOffset += upcomingleftOffset
+            if newRowMetadata.leftIncDec > 0 && row > 0 {
+                for prevRow in (0...(row-1)).reversed() {
+                    allRowsMetaData[prevRow].leftOffset += newRowMetadata.leftIncDec
+                }
+            }
+            if newRowMetadata.leftIncDec < 0 {
+                upcomingleftOffset += abs(newRowMetadata.leftIncDec)
+                newRowMetadata.transRowLeftOffset += upcomingleftOffset
+            }
+            allRowsMetaData.append(newRowMetadata)
+        }
+        return(allRowsMetaData)
+    }
+
+    public func makeRowMetadata(stitchRow: [String], rowNumber: Int) -> RowInfo {
+        var rowData = RowInfo(row: stitchRow, rowNumber: rowNumber)
+        rowData.leftIncDec = findLeftChanges(stitchRow: stitchRow)
+        rowData.rightIncDec = findRightChanges(stitchRow: stitchRow)
+        rowData.width = stitchRow.count
+        if rowNumber == 0 {
+            rowData.bottomLine = ChartConstructor().makeBottomRow(width: rowData.width)
+        } else {
+            rowData.bottomLine = ChartConstructor().makeMiddleRow(width: rowData.width,
+                                                                  rDiff: rowData.rightIncDec,
+                                                                  lDiff: rowData.leftIncDec)
+        }
+        rowData.stitchSymbols = ChartConstructor().makeStitchRow(row: rowData.row)
+        return(rowData)
+    }
+
+    public func findCenter(stitchRow: [String]) -> Int {
+        return(stitchRow.count/2)
+    }
+
+    public func findLeftChanges(stitchRow: [String]) -> Int {
+        let center = findCenter(stitchRow: stitchRow)
+        var totalLeftChange = 0
+        for stitch in 0..<(center) {
+            let lookupStitch = allowedStitchesInfo.first(where: { $0.name == stitchRow[stitch] })
+            totalLeftChange += lookupStitch!.incDecValue
+        }
+        return(totalLeftChange)
+    }
+
+    public func findRightChanges(stitchRow: [String]) -> Int {
+        let center = findCenter(stitchRow: stitchRow)
+        var totalRightChange = 0
+        for stitch in center..<(stitchRow.count) {
+            let lookupStitch = allowedStitchesInfo.first(where: { $0.name == stitchRow[stitch] })
+            totalRightChange += lookupStitch!.incDecValue
```

Comment:
Given this duplicated code here, perhaps you can extract it into a private function.